torch/fx/experimental/subgraph_creation_example.py

```diff
@@ -0,0 +1,212 @@
+
+            if node.op not in ['call_module', 'get_attr']:
+                target = node.target
+            else:
+                target_atoms = node.target.split('.')
+                target_attr = m
+                for atom in target_atoms:
+                    if not hasattr(target_attr, atom):
+                        raise RuntimeError(f'Operator target {node.target} not found!')
+                    target_attr = getattr(target_attr, atom)
+                partition.targets[node.target] = target_attr
+                target = target_atoms[-1]
+
+            new_node = partition.graph.create_node(op=node.op, target=target, args=gathered_args,  # type: ignore 
+                                                   kwargs=gathered_kwargs)  # type: ignore  
+            partition.environment[node] = new_node
+
+    # Set up values to construct base module
+    base_mod_env : Dict[str, Node] = {}
+    base_mod_graph : Graph = Graph()
+    base_mod_attrs : Dict[str, GraphModule] = {}
+    for node in m.graph.nodes:
+        if node.op == 'placeholder':
+            base_mod_env[node.name] = base_mod_graph.placeholder(node.name)
+        elif node.op == 'get_attr':
+            base_mod_env[node.name] = base_mod_graph.get_attr(node.target)
+            attr_val = m
+            for atom in node.target.split('.'):
+                if not hasattr(attr_val, atom):
+                    raise RuntimeError(f'Node target {node.target} not found!')
+                attr_val = getattr(attr_val, atom)
+            base_mod_attrs[node.target] = attr_val
+
+    # Do some things iterating over the partitions in topological order again:
+    # 1) Finish off submodule Graphs by setting corresponding outputs
+    # 2) Construct GraphModules for each submodule
+    # 3) Construct the base graph by emitting calls to those submodules in
+    #    topological order
+
+    for partition_name in sorted_partitions:
+        partition = partitions[partition_name]
+
+        # Set correct output values
+        output_vals = tuple(partition.environment[orig_nodes[name]] for name in partition.outputs)
+        partition.graph.output(output_vals)
+
+        # Construct GraphModule for this partition
+        submod_name = f'submod_{partition_name}'
+        base_mod_attrs[submod_name] = GraphModule(partition.targets, partition.graph)
+
+        # Emit call in base graph to this submodule
+        output_args : Optional[Tuple[torch.fx.node.Argument, ...]] = tuple([base_mod_env[name] for name in partition.inputs])
+        output_val = base_mod_graph.call_module(submod_name, output_args)
+        if len(partition.outputs) > 1:
+            # Unpack multiple return values from submodule
+            output_val_proxy = torch.fx.proxy.Proxy(output_val)
+            for i, output_name in enumerate(partition.outputs):
+                base_mod_env[output_name] = output_val_proxy[i].node  # type: ignore
```

Comment:
The last mypy error isn't realted to the first three, the Proxy class doesn't like to be indexed

mypy error: Value of type "Proxy" is not indexable [index]

Reply:
See if this works?

https://gist.github.com/jamesr66a/642a92a473936487cb33bd424a45f35a

Otherwise I'm not sure, and it's probably okay to keep the ignore annotation

Reply:
Yeah it's the same issue with the changes to proxy, I kept the ignore annotation